js/index.js

```diff
@@ -135,44 +135,45 @@ messageForm.addEventListener("submit", function (event) {
   messageForm.reset();
 });
 
-// Fetch GitHub Repositories
-const githubRequest = new XMLHttpRequest();
-
-githubRequest.open("GET", "https://api.github.com/users/DoinaFitchevici/repos");
-
-// Handle Response from Server
-
-githubRequest.addEventListener("load", (event) => {
-  const repositories = JSON.parse(githubRequest.response);
-  console.log(repositories);
-
-  // Display Repositories in List
-  const projectSection = document.getElementById("projects");
-  const projectList = projectSection.querySelector("ul");
-  for (let i = 0; i < repositories.length; i++) {
-    const project = document.createElement("li");
-
-    // Transform your repository names into <a> tags that link to GitHub
-    const link = document.createElement("a");
-    link.href = repositories[i].html_url;
-    link.innerText = repositories[i].name;
-    project.appendChild(link);
-
-    // Display additional information about your repositories (i.e. description, date, etc.)
-    const repositoryDescription = document.createElement("p");
-    repositoryDescription.textContent = repositories[i].description;
-    project.appendChild(repositoryDescription);
-
-    const repositoryDate = document.createElement("p");
-    const createdAt = new Date(repositories[i].created_at);
-    const formattedDate = createdAt.toLocaleDateString(undefined, {
-      year: "numeric",
-      month: "long",
-      day: "numeric",
-    });
-    repositoryDate.textContent = "Created at: " + formattedDate;
-    project.appendChild(repositoryDate);
-    projectList.appendChild(project);
-  }
-});
-githubRequest.send();
+// Using the Fetch API, create a "GET" request to the same GitHub API url as before
+//  hint: the fetch function
+//  hint: "GET" is the default method for fetch
+fetch("https://api.github.com/users/DoinaFitchevici/repos", {
+  method: "GET",
+})
+  // Chain a then method to your fetch call and pass it a function that returns the response JSON data
+
+  .then((response) => response.json())
+  // Chain another then method and pass it a function, inside of which you can paste the code from your previous "load" event listener function
+
+  .then((repositories) => {
+    const repoSection = document.getElementById("projects");
+    const repoList = repoSection.querySelector("ul");
+    for (let i = 0; i < repositories.length; i++) {
+      // console.log(repositories[i]);
+      const repository = document.createElement("li");
+      // repository.innerText = repositories[i].name;
+      let repositoryContent = `
+            <p><a href='${repositories[i].html_url}' target="_blank">${
+        repositories[i].name
+      }</a></p>
+            ${
+              repositories[i].description
+                ? `<p>${repositories[i].description}</p>`
```

Comment:
One thing to consider: What would happen if the description contained a `<` character? Even though it's unlikely, if it were to happen, it would mess up the display of the page. There are different ways to address this:
- Use DOM methods to build the content.
- HTML escape the values you insert.
- Use a framework like React that handles the escaping for you, by default.

Reply:
(The same problem comes up in other contexts: When working with databases, we need to be careful about sql injection attacks.)